### DISCUSSION TITLE
求问 layui table 如何实现 search 功能

### DISCUSSION THREAD

**Discussion by Laiyiwen-02** [layui/layui#2161] [问答] [answered]
如题，看官网文档有关实现表格搜索的说明太少了，并且很多 csdn 上的文章也过于古老，有无大佬帮助给下 js 写法。

**Comment by LSL1618** [comment]
表格搜索不是应该通过后端接口传递搜索条件来实现吗？

**Reply by Laiyiwen-02** [reply]
额你可能没理解我的意思就是说假设我们现在有一个 input 和一个 table，如何实现在 table 中获取相关信息，然后用 layui 实现

**Reply by hulang** [reply]
> 额你可能没理解我的意思就是说假设我们现在有一个 input 和一个 table，如何实现在 table 中获取相关信息，然后用 layui 实现

你这发的问题，1没有代码，2没有截图，让大家猜吗？
是静态数据的话，用：var data = table.cache['test'];，循环data进行去匹配你需要的结果

**Reply by Laiyiwen-02** [reply]
就是静态数据，问题是有没有一个靠谱的搜索函数

**Reply by Laiyiwen-02** [reply]
我去截一下图

**Reply by LSL1618** [reply, marked as answer]
不要局限于table组件初始化渲染方法，可以结合其他自带的方法达到你的要求，首先你的表格内容能显示出来说明已经有数据了，那就要找找如何拿到这个数据，`table.getData()、table.cache`都可以拿到，再从`input`拿到你的搜索关键字，然后用关键字去循环匹配拿到的表格数据，其实就是一个筛选过滤的过程，至于用什么字段(一个或者多个字段)如何匹配(大于、小于、等于、包含等等)取决于你的搜索要求，接着得到目标数据后进行表格数据重载`table.reloadData()`就完事了。

**Reply by Laiyiwen-02** [reply]
会了，谢谢。  
我的数据本来就是用 php 传输的，你这么一说，大概想到了就是说可以用 GET 来弄。  
谢谢，此帖结！
